src/arrayMethodSort.js

```diff
@@ -4,8 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    const copyArr = [...this];
+    let defaultFn = compareFunction;
+
+    if (!compareFunction) {
+      defaultFn = (a, b) => String(a) > String(b);
+    }
+
+    for (let i = 0; i < copyArr.length - 1; i++) {
+      for (let j = 0; j < copyArr.length - 1 - i; j++) {
+        if (defaultFn(copyArr[j], copyArr[j + 1]) > 0) {
+          [copyArr[j], copyArr[j + 1]] = [copyArr[j + 1], copyArr[j]];
+        }
+      }
+    }
```

Comment:
While the bubble sort algorithm is correctly implemented, it is not the most efficient sorting algorithm for larger arrays. Consider researching more efficient sorting algorithms like quicksort or mergesort for better performance on larger datasets.